xray/services/enrich.go

```diff
@@ -0,0 +1,99 @@
+package services
+
+import (
+	"encoding/json"
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"net/http"
+)
+
+const (
+	importGraph    = "api/v1/scan/import"
+	importGraphXML = "api/v1/scan/import_xml"
+)
+
+func (ss *ScanService) ImportGraph(scanParams XrayGraphImportParams) (string, error) {
+	httpClientsDetails := ss.XrayDetails.CreateHttpClientDetails()
+	var v interface{}
+	err := json.Unmarshal(scanParams.SBOMInput, &v)
+	utils.SetContentType("application/json", &httpClientsDetails.Headers)
+	url := ss.XrayDetails.GetUrl() + importGraph
+	if err != nil {
+		utils.SetContentType("application/xml", &httpClientsDetails.Headers)
+		url = ss.XrayDetails.GetUrl() + importGraphXML
```

Comment:
please add comment why this is a fallback

Reply:
added